test/mocha/27-http-edv-client.js

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright (c) 2018-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+require('bedrock-edv-storage');
+const bedrock = require('bedrock');
+const brEdvStorage = require('bedrock-edv-storage');
+const {util: {clone}} = bedrock;
+const {config} = bedrock;
+const helpers = require('./helpers');
+const assertions = require('./assertions');
+const mockData = require('./mock.data');
+const {CapabilityAgent} = require('webkms-client');
+let actors;
+let urls;
+
+describe('bedrock-edv-storage HTTP API - edv-client', () => {
+  let passportStub;
+
+  before(async () => {
+    await helpers.prepareDatabase(mockData);
+    actors = await helpers.getActors(mockData);
+    // common URLs
+    const {baseUri} = config.server;
+    const root = `${baseUri}/edvs`;
+    const invalid = `${baseUri}/edvs/invalid`;
+    urls = {
+      edvs: root,
+      invalidDocuments: `${invalid}/documents`,
+      invalidQuery: `${invalid}/query`
+    };
+  });
+
+  before(() => {
+    passportStub = helpers.stubPassport({actor: actors['alpha@example.com']});
+  });
+
+  after(() => {
+    passportStub.restore();
+  });
+
+  describe('update API', () => {
+    let capabilityAgent;
+    let edvClient;
+    let accounts;
+    const mockEdvId = `${config.server.baseUri}/edvs/z19xXoFRcobgskDQ6ywrRaa12`;
+    before(async () => {
+      await helpers.prepareDatabase(mockData);
+      actors = await helpers.getActors(mockData);
+      accounts = mockData.accounts;
+    });
+    before(async () => {
+      const actor = actors['alpha@example.com'];
+      const account = accounts['alpha@example.com'].account;
+      const edvConfig = {...mockData.config, controller: account.id};
+      edvConfig.id = mockEdvId;
+      await brEdvStorage.insertConfig({actor, config: edvConfig});
+    });
+    before(async () => {
+      const secret = 'e4f2f31d-b21a-427e-b49a-b4447d5bb219';
+      const handle = 'testKey4';
+      capabilityAgent = await CapabilityAgent.fromSecret(
+        {secret, handle});
+
+      const keystoreAgent = await helpers.createKeystore({capabilityAgent});
+
+      // corresponds to the passport authenticated user
+      const actor = actors['alpha@example.com'];
+
+      ({edvClient} = await helpers.createEdv(
+        {actor, capabilityAgent, keystoreAgent, urls}));
+    });
+
+    for(const test of helpers.sequenceNumberTests({update: true})) {
+      it(test.title, async () => {
+        let result;
+        let err;
+        const actor = actors['alpha@example.com'];
+        const doc = clone(mockData.docWithNoIndexedOrRecipients);
+        doc.id = await helpers.generateRandom();
+        doc.sequence = test.sequence;
```

Comment:
I think the document inserted here should have a sequence of `test.sequence - 1` so that when the client increments the sequence number during an update it's testing the actual target value.
```suggestion
        doc.sequence = test.sequence - 1;
```